src/GUI/Shared/Helpers/RoslynEntityPropertyRenamer.cs

```diff
@@ -0,0 +1,146 @@
+            {
+                /* If existing project file fails to load, we can fallback to creating an adhoc workspace with an in memory project
+                 * that contains only the cs files we are concerned with.
+                 *
+                 * Currently, this will successfully rename the model properties but it does NOT process the references from the Configuration
+                 * classes.  This is likely because Roslyn is not resolving information about EntityTypeBuilder<> such that it
+                 * can identify the model property references.
+                 *
+                 * This issue *should* be resolved simply by adding EFC references to the project (as is illustrated below) but
+                 * it's still not working.
+                 *
+                 * To do: resolve paths to EFC assemblies and use those in the refAssemblies list below.
+                 */
+                var cSharpFolders = new HashSet<string>();
+                if (contextFolder?.Length > 0 && Directory.Exists(Path.Combine(dir, contextFolder)))
+                {
+                    cSharpFolders.Add(Path.Combine(dir, contextFolder));
+                }
+
+                if (modelsFolder?.Length > 0 && Directory.Exists(Path.Combine(dir, modelsFolder)))
+                {
+                    cSharpFolders.Add(Path.Combine(dir, modelsFolder));
+                }
+
+                if (cSharpFolders.Count == 0)
+                {
+                    return 0;
+                }
+
+                var cSharpFiles = cSharpFolders
+                    .SelectMany(o => Directory.GetFiles(o, "*.cs", SearchOption.AllDirectories))
+                    .Distinct().ToList();
+                if (cSharpFiles.Count == 0)
+                {
+                    return 0;
+                }
+
+                var refAssemblies = new[]
+                {
+                    // typeof(IEntityTypeConfiguration<>).Assembly, typeof(SqlServerValueGenerationStrategy).Assembly,
+                    // typeof(Microsoft.EntityFrameworkCore.SqlServer.Query.Internal.SearchConditionConvertingExpressionVisitor).Assembly,
+                    // typeof(Microsoft.EntityFrameworkCore.RelationalDbFunctionsExtensions).Assembly,
+                    typeof(NotMappedAttribute).Assembly,
+                }.Distinct().ToList();
+                var refs = refAssemblies.Select(o => MetadataReference.CreateFromFile(o.Location)).ToList();
+
+                try
+                {
+                    var workspace = cSharpFiles.GetWorkspaceForFilePaths(refs);
+                    project = workspace.CurrentSolution.Projects.First();
+                }
+                catch
+                {
+                    // TODO respond to this from caller
+                    return -1; // log this?
+                }
+            }
+
+            var rulesText = File.ReadAllText(path);
+            var rulesList = RenamingRulesSerializer.TryRead(rulesText);
```

Comment:
TryRead is expecting a path here.  Pass "path" instead of "rulesText" and remove previous line